fs-storage/src/base_storage.rs

```diff
@@ -1,27 +1,37 @@
 use data_error::Result;
 use std::collections::BTreeMap;
 
+#[derive(Debug, PartialEq, PartialOrd, Ord, Eq, Clone)]
+/// Represents the synchronization status of the storage.
+pub enum SyncStatus {
+    /// No synchronization needed.
+    NoSync,
+    /// Load key-value mapping from the file system.
+    UpSync,
+    /// Write key-value mapping to the file system.
+    DownSync,
+    /// Changes from both in-memory and file system need to be merged before syncing.
+    FullSync,
+}
```

Comment:
Probably I would reword names of these values like "InSync", "Ours", "Theirs", "Both" to accent on which side should be updated. This would also make the enum potentially re-usable with other kinds of backend (filesystem is only starting point, later we might use Iroh directly).

Reply:
```
    /// No synchronization needed.
    InSync,
    /// In-memory key-value mapping is stale.
    MappingStale,
    /// External file system storage is stale.
    StorageStale,
    /// In-memory key-value mapping and external file system storage diverge.
    Diverge,
```

How about this? Ours and theirs can become confusing. Since this dealing with FileStorage, mapping and storage makes it clear what we're refering to.

Reply:
This looks good. Ours/theirs is kinda conventional but I agree that it can be confusing.